facebookincubator/create-react-app
# Whitelist files that can be embedded through url-loader

Change the current catch-all loader to use `file-loader` instead of `url-loader`, and exclude common image file extensions. Add another `url-loader` for images, configured identically to the original catch-all loader.

Part of issue #1293.

## Testing

1. Create a demo app with `./tasks/cra.sh my-app-test`
2. `cd my-app-test`
3. Add a 1K PNG, a 50K PNG and a PDF to `./src`
4. Import each resource into `App.js` and include them in the page
```
import smallPicture from './small.png';
import largePicture from './large.png';
import someDoc from './someDoc.pdf';

// ...
<p><img src={smallPicture} alt="small" /></p>
<p><img src={largePicture} alt="large" /></p>
<p><a href={someDoc}>Here's a PDF</a></p>
```
5. Start dev mode and check that:
   * The small image is embedded
   * The large image is referenced
   * The PDF is referenced
```
yarn start
```
7. Build a production bundle:
```
yarn run build
```
8. Check that the production build works as expected by serving it and browsing [http://localhost/](http://localhost/)
```
docker run -it --rm -v $PWD/build:/usr/share/nginx/html -p 80:80 nginx:alpine
```
